shells/backends/gcp_tf_backend/src/driver.py

```diff
@@ -101,195 +101,58 @@ def _generate_state_file_string(self, gcp_backend_resource: GcpTfBackend):
+            os.environ[GOOGLE_APPLICATION_CREDENTIALS] = DYNAMIC_JSON
+        # Keys not defines on GCP TF BACKEND RESOURCE (CLP reference should have been set)
+        else:
+            # CLP had not been set...
+            if not gcp_backend_resource.cloud_provider:
+                self._raise_and_log(logger, "At least one method of authentication should be filled")
+            # Check a correct CLP has been reference
+            clp_details = api.GetResourceDetails
+            clp_resource_details = self._get_and_validate_clp(clp_details, gcp_backend_resource, logger)
+            if clp_resource_details.ResourceModelName == GCP1G_MODEL:
+                myactual=clp_details(clp_resource_details.Name)
+                clp_json_path = self._fill_backend_sercret_vars_data(myactual, GCP1G_MODEL)
+                os.environ[GOOGLE_APPLICATION_CREDENTIALS] = clp_json_path
+        # except Exception as e:
+        #     self._raise_and_log(logger, f"There was an issue accessing GCP. {e}")
+        bucket_name = gcp_backend_resource.bucket_name
+        if bucket_name:      
+            self._validate_bucket_exists(bucket_name, context, logger)
 
     def _fill_backend_sercret_vars_data(self, myactual, clp):
-        print(myactual)
         for attr in  myactual.ResourceAttributes:
             if attr.Name == "Google Cloud Provider.Credentials Json Path":
                 clp_json_path = attr.Value
-        # gcp_model_prefix = ""
-        # self._backend_secret_vars = {}
-        # for attr in clp_resource_attributes:
-        #     if attr.Name == gcp_model_prefix + "CREDENTIALS JSON PATH":
-        #         json_path_new = attr.Value
                 return clp_json_path
-                # self._backend_secret_vars["subscription_id"] = attr.Value
-            # if attr.Name == azure_model_prefix + "Azure Tenant ID":
-            #     self._backend_secret_vars["tenant_id"] = attr.Value
-            # if attr.Name == azure_model_prefix + "Azure Application ID":
-            #     self._backend_secret_vars["client_id"] = attr.Value
-            # if attr.Name == azure_model_prefix + "Azure Application Key":
-            #     dec_client_secret = api.DecryptPassword(attr.Value).Value
-            #     self._backend_secret_vars["client_secret"] = dec_client_secret   
 
-    # def _create_clp_gcp_session(self, json_path):
-    #     os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = json_path
 
-
-    def _validate_clp(self, api, gcp_backend_resource, logger):
-        clp_resource_name = gcp_backend_resource.cloud_provider
-        clp_details = api.GetResourceDetails(clp_resource_name)
-        clp_res_model = clp_details.ResourceModelName
-        clpr_res_fam = clp_details.ResourceFamilyName
+    def _get_and_validate_clp(self, clp_details, gcp_backend_resource, logger):
+        # clp_resource_name = gcp_backend_resource.cloud_provider
+        clp_details_resource_name = clp_details(gcp_backend_resource.cloud_provider)
```

Comment:
I would recommend change the name of this variable to "clp_details_resource" or "clp_details_resource_info". Because clp_details_resource_name implies that it's only the resource name which it is not

Reply:
ok, good point.